packages/java/plugin-base/src/test/java/dev/hilla/plugin/base/HillaAppInitUtilityTest.java

```diff
@@ -158,18 +162,50 @@ public void withCorrectLitDependencies_scaffoldForLitSucceeds()
                 .toFile().exists());
         Assertions.assertTrue(projectDirectory
                 .resolve("frontend/views/main-view.ts").toFile().exists());
+
+        assertEndpointIsNotGeneratedForInvalidPackages();
+    }
+
+    private void assertEndpointIsNotGeneratedForInvalidPackages() {
+        Assertions.assertFalse(projectDirectory.resolve(
+                "src/main/java/my/package1/app/endpoints/HelloEndpoint.java")
+                .toFile().exists());
+        Assertions.assertFalse(projectDirectory.resolve(
+                "src/main/java/my/single/line/comment/endpoints/HelloEndpoint.java")
+                .toFile().exists());
+        Assertions.assertFalse(projectDirectory.resolve(
+                "src/main/java/my/javadoc/package1/endpoints/HelloEndpoint.java")
+                .toFile().exists());
+        Assertions.assertFalse(projectDirectory.resolve(
+                "src/main/java/my/package1/in/string/literal/endpoints/HelloEndpoint.java")
+                .toFile().exists());
     }
 
     private void createSpringBootApplicationClass() throws IOException {
 
         var content = """
-                package my;
+                /*
+                * Some comments that might also contains a package
+                * declaration statement:
+                * package my.package1.app;
+                */
```

Comment:
Let's test this case also:
```suggestion
                /*
                Some comments that might also contains a package
                declaration statement:
                package my.package1.app;
                */
```